examples/computecloth/computecloth.cpp

```diff
@@ -306,26 +303,68 @@ class VulkanExample : public VulkanExampleBase
 
 			// Dispatch the compute job
 			const uint32_t iterations = 64;
+			static_assert(iterations % 2 == 0, "The below code assumes an even number of iterations.");
+			uint32_t readSet{ 0 };
 			for (uint32_t j = 0; j < iterations; j++) {
-				readSet = 1 - readSet;
```

Comment:
OK, I get the two command buffers now.  This was previously a global variable and because this is at the start of the loop, on the first iteration we read from the output buffer, which is what's populated by the original `prepareStorageBuffers`.  The second command buffer however would invert that.

However, since `readSet` is never modified outside of this function, during the actual render loop we only ever use one of the two command buffers.

I think having one compute command buffer and having it always treat the input and output buffers in the same way each frame is less error prone.